maintenance_plan_meter_mode/models/maintenance.py

```diff
@@ -53,17 +64,24 @@ def _onchange_meter_mode(self):
             self.planning_step = None
         else:
             self.meter_current_value = None
+            self.meter_autoinc_mode = None
             self.meter_autoinc_value = None
             self.meter_autoinc_unit = None
             self.meter_autoinc_maxvalue = None
             self.meter_autoinc_lastvalue = None
             self.meter_autoinc_lastupdate = None
 
+    @api.onchange('meter_autoinc_mode')
+    def _onchange_meter_mode(self):
+        if self.meter_autoinc_mode:
+            self.meter_autoinc_value = None
```

Comment:
The suggested code is actually the same, but maybe clearer (coherent with cron recordset selection domain)
```suggestion
            self.meter_autoinc_value = 0
```